src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/PostClaimOrchestrationHandler.java

```diff
@@ -36,33 +66,75 @@ public PostClaimOrchestrationHandler(
+                ? uploadOperationService.uploadDocument(claim, event.getAuthorisation(), docs.getSealedClaim(),
+                CaseEvent.SEALED_CLAIM_UPLOAD)
+                : claim;
+
+        uploadClaimIssueReceiptOperation = (claim, event, docs) ->
+            isUploadClaimReceiptSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? uploadOperationService.uploadDocument(claim, event.getAuthorisation(), docs.getClaimIssueReceipt(),
+                CaseEvent.CLAIM_ISSUE_RECEIPT_UPLOAD)
+                : claim;
+
+        rpaOperation = (claim, event, docs) ->
+            isRpaOperationSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? rpaOperationService.notify(claim, event.getAuthorisation(), docs.getSealedClaim())
+                : claim;
+
+        notifyStaffOperation = (claim, event, docs) ->
+            isNotifyStaffSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? notifyStaffOperationService.notify(claim, event.getAuthorisation(), docs.getSealedClaim(),
+                docs.getDefendantLetter())
+                : claim;
+
+        notifyCitizenOperation = (claim, event, docs) ->
+            isNotifyCitizenSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? claimantOperationService.notifyCitizen(claim, event.getSubmitterName(), event.getAuthorisation())
+                : claim;
+
+        notifyRepresentativeOperation = (claim, event, docs) ->
+            isNotifyCitizenSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? claimantOperationService.confirmRepresentative(claim, event.getSubmitterName(),
+                ((RepresentedClaimCreatedEvent) event).getRepresentativeEmail(), event.getAuthorisation())
+                : claim;
     }
 
     @EventListener
     public void citizenIssueHandler(CitizenClaimCreatedEvent event) {
         try {
             Claim claim = event.getClaim();
             String authorisation = event.getAuthorisation();
-            String submitterName = event.getSubmitterName();
-
-            Claim updatedClaim = pinOrchestrationService.process(claim, authorisation, submitterName);
+            Claim updatedClaim;
 
-            PDF sealedClaimPdf = documentOrchestrationService.getSealedClaimPdf(claim);
-            updatedClaim = uploadOperationService.uploadDocument(updatedClaim, authorisation, sealedClaimPdf);
+            GeneratedDocuments generatedDocuments =
+                documentOrchestrationService.generateForCitizen(claim, authorisation);
 
-            PDF claimIssueReceiptPdf = documentOrchestrationService.getClaimIssueReceiptPdf(claim);
-            updatedClaim = uploadOperationService.uploadDocument(updatedClaim, authorisation, claimIssueReceiptPdf);
+            updatedClaim = CompletableFuture
+                .supplyAsync(() -> performPinOperations.perform(claim, event, generatedDocuments))
```

Comment:
doesn't seems to achieve any benefit by using completable future here, simply supplier is called in separate thread, however main thread will wait for it's completion. And output from fist call is fed in next call.  This overall shows they are just synchronous calls. One shouldn't use completable future here, Please justify if you have different opinion

Reply:
Completable Future is not for Async Operation. Future will become blocking in one stage when you are getting the value from it. Here all the supplier will be run in separate thread. With having multi core machines, spawning threads is not a NO GO.  Completable Future can be utilised for non-blocking as well and parallel processing with JOIN as well. Here used to make use of multi threading to achieve suppliers to run in separate thread.